python_modules/dagster/dagster/_core/storage/partition_status_cache.py

```diff
@@ -270,17 +289,22 @@ def _build_failed_partition_subset(
         )
         if new_failed_partitions
         else partitions_def.empty_subset(),
+        partitions_def.empty_subset().with_partition_keys(
+            get_validated_partition_keys(instance, partitions_def, in_progress_partitions)
```

Comment:
I think after https://github.com/dagster-io/dagster/pull/12897 was merged you'll need to rebase onto master. And I think the first argument should be `dynamic_partitions_store` (which caches dynamic partitions, ensuring that we only fetch dynamic partitions once per def)